src/context.c

```diff
@@ -473,25 +473,63 @@ ly_ctx_get_options(const struct ly_ctx *ctx)
 API LY_ERR
 ly_ctx_set_options(struct ly_ctx *ctx, uint16_t option)
 {
+    LY_ERR lyrc;
     LY_CHECK_ARG_RET(ctx, ctx, LY_EINVAL);
     LY_CHECK_ERR_RET(option & LY_CTX_NO_YANGLIBRARY, LOGARG(ctx, option), LY_EINVAL);
+    lyrc = LY_SUCCESS;
+
+    if (!(ctx->flags & LY_CTX_SET_PRIV_PARSED) && (option & LY_CTX_SET_PRIV_PARSED)) {
+        ctx->flags |= LY_CTX_SET_PRIV_PARSED;
+        if ((lyrc = lys_recompile(ctx, 0))) {
+            ly_ctx_unset_options(ctx, LY_CTX_SET_PRIV_PARSED);
+        }
+    }
 
     /* set the option(s) */
-    ctx->flags |= option;
+    if (!lyrc) {
+        ctx->flags |= option;
+    }
+
+    return lyrc;
+}
 
+static LY_ERR
+lysc_module_priv2null(struct lysc_node *node, void *UNUSED(data), ly_bool *UNUSED(dfs_continue))
```

Comment:
The custom is to use the `x2y` notation for conversion functions, that is not the case here, so please keep that in mind. Also, I prefer every callback to be clearly marked that way so I prefer a suffix `_cb`. So you could rename this to `lysc_node_clear_priv_dfs_cb` or something similar.

Reply:
ok